recipe/meta.yaml

```diff
@@ -10,18 +10,17 @@ source:
     sha256: c3e5e9fdd5004dcb542feda5ee4f0ff0744628baf8ed2dd5d66f8ca1197cb1a1
 
 build:
-    number: 0
-    features:
-        - vc9  # [win and py27]
-        - vc10  # [win and py34]
-        - vc14  # [win and py>=35]
-    skip: true  # [win and py>35]
+    number: 2
+    run_exports:
+        #    https://abi-laboratory.pro/tracker/timeline/zlib/
+        - {{ pin_subpackage('zlib', max_pin='x.x.x') }}
```

Comment:
The past 4 releases have been fine.  Why pin so tightly?  With zlib so low in the stack, it'll be very painful to rebuild with the next version - especially if it's actually totally compatible and would otherwise be fine.

Our experience with this has been that it is better to give packages the benefit of the doubt.  You can either optimize for users never breaking (pin tightly, like here), or optimize for package builder time (pin as loosely as reasonably achievable).  Since you'd have to rebuild if something breaks, anyway, you're primarily just making work for yourself by pinning tightly.

Reply:
As an example of the types of problems seen, please [this diff report between 1.2.8 and 1.2.9]( https://abi-laboratory.pro/tracker/compat_report/zlib/1.2.8/1.2.9/d8c18/abi_compat_report.html#Type_Problems_Low ). Note they changed a `struct` and it will affect public API functions. This happened in a relatively late patch release. Admittedly 1.2.10 merely changed the copyright and 1.2.11 was compatible with 1.2.10. However these were released within days of each other. The history before 1.2.8 is more unsettling. Also please see [this discussion]( https://github.com/conda-forge/conda-forge.github.io/pull/397#pullrequestreview-45296088 ) for more details.

Reply:
That issue does not account for forward compatibility, which has been the trend.  Comparing the history before 1.2.8 isn't really enlightening, because it comes from 2013.  Current development has all been fine.  Adding symbols is OK.  That's the nature of forward compatibility.